app/dojo.py

```diff
@@ -248,3 +251,113 @@ def print_unallocated(self,filename=None):
 			txt_file.write(output)
 			txt_file.close()
 			return ("\nData has been successfully saved to {}.txt\n" .format(filename))
+
+	def reallocate_person(self, emailaddress, new_roomname):
+		"""This method reallocates a person using their unique identifier,
+		in this case their email address, to a the specified new room
+
+		"""
+		new_room = None
+		person_reallocating = None
+		
+		#check if a person with the email address exists
+		for member in itertools.chain(self.all_staff,self.all_fellows):
+			if member.email == emailaddress:
+				person_reallocating = member
+
+		if person_reallocating is None:
+			return (text_format.CRED + "\nCould not find person with email {}!\n"
+				.format(emailaddress)
+				+ text_format.CEND)
+
```

Comment:
Make use of list comprehensions instead of for loops. As you can see, for loops are pushing you to the right and making the code verbose. example:
`persons = self.all_staff + self.all_fellows`
`if email address in [person.email for person in persons] `
               `person_reallocating = member`
else:
     `return 'some messgae'`
             `